src/ui/citation-popup/view.js

```diff
@@ -1,22 +1,125 @@
+/* 
+For some reason, Joplin does not load the scripts in order.
+This strange behavior was causing the current script to load before he.min.js
+resulting in "he is not defined" error.
+The quick and dirty way to solve this bug is to make sure the current script
+waits until all the other scripts get loaded by Joplin,
+and then starts doing its job
+*/
+setTimeout(main, 5, 10);
+
 /* UI Elements */
-let referenceIdInput = document.getElementById("reference_id");
-let refs = document.getElementsByTagName("li");
-
-/* Event Listeners */
-
-// When selecting a certain reference, set the "reference_id" field accordingly
-let selectedReferenceView = null;
-for (let i = 0; i < refs.length; i++) {
-    let ref = refs[i];
-    ref.addEventListener("click", event => {
-        referenceIdInput.value = event.target.id;
-        
-        // Attach some css style
-        if (selectedReferenceView !== null) {
-            selectedReferenceView.className = "";
-        }
+const inputRefsView = document.getElementById("json");
+const selectedRefsView = document.getElementById("selected_refs_list");
+const output = document.getElementById("output");
+
+function main () {
+
+    /* State */
+    const refs = JSON.parse( he.decode(inputRefsView.textContent) );
+    const invertedRefsIndex = {};
+    refs.forEach(ref => invertedRefsIndex[ ref["id"] ] = ref);
+    const state = {
+        selectedRefs: new Set()
+    };
+
+    configAutoComplete();
 
-        selectedReferenceView = event.target;
-        selectedReferenceView.className = "selected";
+    /* Event Listeners */
+    selectedRefsView.addEventListener("click", event => {
+        if (event.target.classList.contains("icon_remove")) {
+            removeReference(event.target.parentNode.id);
+        }
     });
+
+    function configAutoComplete () {
+
+        const autoCompleteJS = new autoComplete({
+            placeHolder: "Search for references...",
+            data: {
+                src: refs,
+                keys: ["title"],
+                filter: list => {
+                    return list.filter(item => !state.selectedRefs.has(item.value["id"]));
+                }
+            },
+            resultsList: {
+                noResults: true,
+                maxResults: 15,
+                tabSelect: true
+            },
+            resultItem: {
+                element: (item, data) => {
+                    // Modify Results Item Style
+                    item.style = "display: flex; justify-content: space-between;";
+                    // Modify Results Item Content
+                    item.innerHTML = `
+                        <span style="text-overflow: ellipsis; white-space: nowrap; overflow: hidden;">
+                            ${ data.match }
```

Comment:
Shouldn't you escape this too?

Reply:
OH, I forgot to mention that `data.match` is a DOM element that's responsible for highlighting the parts of the title that match the search query, so we can assume that it does escaping by default (I guess because it uses `textContent` internally). Also, it didn't break when I used the title 'This is an <art".